sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/ThriftServerWithSparkContextSuite.scala

```diff
@@ -79,6 +84,45 @@ trait ThriftServerWithSparkContextSuite extends SharedThriftServer {
         "java.lang.NumberFormatException: invalid input syntax for type numeric: 1.2"))
     }
   }
+
+  test("SPARK-33526: Add config to control if cancel invoke interrupt task on thriftserver") {
+    withJdbcStatement { statement =>
+      val forceCancel = new AtomicBoolean(false)
+      val listener = new SparkListener {
+        override def onTaskEnd(taskEnd: SparkListenerTaskEnd): Unit = {
+          taskEnd.reason match {
+            case _: TaskKilled =>
+              if (forceCancel.get()) {
+                assert(System.currentTimeMillis() - taskEnd.taskInfo.launchTime < 1000)
+              } else {
+                assert(System.currentTimeMillis() - taskEnd.taskInfo.launchTime >= 2900)
```

Comment:
avoid the sleep accuracy so check 2.9s instead of 3s.

Reply:
hm, how about doing rough comparisons like `running time of forceCancel=false` > `running time of forceCancel=true`? I'm a bit worried that this test can cause test flakiness.

Reply:
Without this PR, the comparison `running time of forceCancel=false` also can langer than `running time of forceCancel=true` due to accuracy.  How about check 2s ? I believe it is stable enough.